alpha/lib/interfaces/IIndexable.php

```diff
@@ -39,4 +39,8 @@ public function setUpdatedAt($time);
 	 */
 	public function indexToSearchIndex();
 	
+	/**
+	 * This function returns the index name to use
+	 */
+	public function getSphinxIndexName();
```

Comment:
This function can be implemented once instead of 13  times in an abstract class that all relevant classes will inherit from , and it will inherit baseObject.
function getSphinxIndexName()
{
  $IndexClass = $this->getIndexObjectName();
   return kSphinxSearchManager::getSphinxIndexName($IndexClass::getObjectIndexName());
}